### ISSUE TITLE
[FEATURE] API endpoint for liveness & readiness probes

### ISSUE THREAD

**Issue by rursprung** [opensearch-project/OpenSearch-Dashboards#3534]
**Is your feature request related to a problem?**
when running OpenSearch Dashboards (or any application, really) in Kubernetes you want/need to have a liveness (which checks whether the node is still alive - if this fails several times in a row a restart will be triggered) and readiness probe (which is used to decide whether traffic can be directed to the node). similar concepts can also exist in other environments.

without security this might be as easy as just querying a central API endpoint which is known to work. but with security in place this can't be done (because the probes don't have authentication enabled).

**What solution would you like?**
an unsecured endpoint should be provided for the readiness probe. it should return HTTP 200 if the node is healthy and an HTTP 500 if not.
possibly a dedicated endpoint could also be provided for the liveness probe (always returning HTTP 200 - as long as the node is running this is good enough, if the node somehow stops this API will also stop working).

these new endpoints should then also be used in the [opensearch helm charts for dashboards](https://github.com/opensearch-project/helm-charts/blob/72078b10070c44e560f78119d79b3930e2013b79/charts/opensearch-dashboards/templates/deployment.yaml#L97-L110) (note: there's currently a bug in there, see https://github.com/opensearch-project/helm-charts/issues/389).

**What alternatives have you considered?**
for the liveness probe a `tcpSocket` probe can be used instead (which just checks if a TCP connection can be attempted on the specified port, w/o doing any HTTP call). this however doesn't work for the readiness probe (as that needs more information about the system).

**Do you have any additional context?**
* see the [k8s documentation on the probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)
* [details on the probes](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes), namely:
  > The diagnostic is considered successful if the response has a status code greater than or equal to 200 and less than 400.
* for OpenSearch the security plugin provides the `/_plugins/_security/health` API endpoint

**Comment by ashwin-pc** [comment]
@kavilla is this relevant to the performance API that we were thinking about? It can come under that umbrella of API's

**Comment by rursprung** [comment]
> [Triage] @rursprung Seems like this is a more general feature of OpenSearch Dashboards instead of the security dashboard plugin, transferring.

i raised it under the dashboards security plugin because the corresponding OpenSearch API is in the security plugin as well. and the endpoint has to be unprotected. but yeah, if it's in OSD itself and the security plugin just allows unsecured access to it that's also fine!

**Comment by ashwin-pc** [comment]
Yeah i agree, I dont see any reason why this needs to be a secured api, but happy to be corrected here. @rursprung i think liveness and readiness can already be determined via the `http://localhost:5601/api/stats` and `http://localhost:5601/api/status` api's. I think the missing piece here is to allow unauthenticated traffic from querying this endpoints. @peternied is there any reason that we cant allow unauthenticated traffic from querying these endpoints?

**Comment by rursprung** [comment]
> i think liveness and readiness can already be determined via the `http://localhost:5601/api/stats` and `http://localhost:5601/api/status` api's.

i don't have access to an OpenSearch Dashboards instance right now and i haven't found the API documentation in the [OpenSearch docs](https://opensearch.org/docs/latest). what do these APIs return? in order to use them as the readiness & liveness probes they'll have to respond with the HTTP status codes as expected by k8s (k8s doesn't care about the response body).
